pkg/syncer/metadatasyncer.go

```diff
@@ -1048,23 +1047,46 @@ func csiUpdatePod(ctx context.Context, pod *v1.Pod, metadataSyncer *metadataSync
 					log.Errorf("failed to get volume id for volume name: %q with err=%v", pv.Name, err)
 					continue
 				}
-				updateSpec := &cnstypes.CnsVolumeMetadataUpdateSpec{
-					VolumeId: cnstypes.CnsVolumeId{
-						Id: volumeHandle,
-					},
-					Metadata: cnstypes.CnsVolumeMetadata{
-						ContainerCluster:      containerCluster,
-						ContainerClusterArray: []cnstypes.CnsContainerCluster{containerCluster},
-						EntityMetadata:        metadataList,
-					},
+			}
+		} else {
+			if volume.VsphereVolume != nil && metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) {
+				// Inline migrated volumes with no PVC
+				var podMetadata *cnstypes.CnsKubernetesEntityMetadata
+				if !deleteFlag {
+					podMetadata = cnsvsphere.GetCnsKubernetesEntityMetaData(pod.Name, nil, deleteFlag, string(cnstypes.CnsKubernetesEntityTypePOD), pod.Namespace, metadataSyncer.configInfo.Cfg.Global.ClusterID, nil)
+				} else {
+					podMetadata = cnsvsphere.GetCnsKubernetesEntityMetaData(pod.Name, nil, deleteFlag, string(cnstypes.CnsKubernetesEntityTypePOD), pod.Namespace, metadataSyncer.configInfo.Cfg.Global.ClusterID, nil)
 				}
-
-				log.Debugf("Calling UpdateVolumeMetadata for volume %s with updateSpec: %+v", updateSpec.VolumeId.Id, spew.Sdump(updateSpec))
-				if err := metadataSyncer.volumeManager.UpdateVolumeMetadata(ctx, updateSpec); err != nil {
-					log.Errorf("UpdateVolumeMetadata failed for volume %s with err: %v", volume.Name, err)
+				metadataList = append(metadataList, cnstypes.BaseCnsEntityMetadata(podMetadata))
+				var err error
+				migrationVolumeSpec := migration.VolumeSpec{VolumePath: volume.VsphereVolume.VolumePath}
+				volumeHandle, err = volumeMigrationService.GetVolumeID(ctx, migrationVolumeSpec)
```

Comment:
Not related to this change, but i see that `volumeMigrationService.GetVolumeID` is called at many places. Can we pass a pointer to `migrationVolumeSpec` as a function argument rather than a copy?

Reply:
@divyenpatel Any suggestions/comments here ?

Reply:
Created an [issue](https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/375) for this. We can address this in a separate PR.